### PR TITLE
Remove qt3 from Table widget in MantidPlot

### LINKED ISSUE #15314: Implement Table using Qt4 widget(s).
This issue is a part of #11891 to deal with Tables.

### PR DESCRIPTION
**Pull request by mantid-roman**
Fixes #15314

Table is refactored to use QTableWidget (Qt4).

**To test**
- Run all the commands in all the menus related to tables. (Excluding cell value calculation which will be dealt with in #15316.
- Check that sample log plotting works.
- Check that imported table workspaces still work.
- Anything else that can test that behaviour of tables didn't change.

### CODE REVIEW

**Comment by martyngigg** [comment]
@mantid-roman This could do with some sort of testing instructions for a few ideas of what to test with the tables

File: MantidPlot/src/Folder.cpp [hunk 1]
```diff
@@ -33,6 +33,8 @@
 #include <QApplication>
 #include <QDateTime>
 
+#include <Q3Header>
```
Comment (martyngigg) [inline MantidPlot/src/Folder.cpp]: This include looks a bit suspicious...

Comment (mantid-roman) [inline MantidPlot/src/Folder.cpp]: Folder will be refactored in #15315.

Comment (martyngigg) [inline MantidPlot/src/Folder.cpp]: :+1:

**Comment by martyngigg** [comment]
The code changes are fine. Moving on to functional testing

**Comment by martyngigg** [comment]
Functional testing:

The following were tried with a random LOQ data set from ISIS:
  - Viewing and plotting sample logs works
  - The detector table can be shown and looks like it is correct.

Table feature testing:
  - Import Table Workspace works
  - All options under the `Table` menu work as expected

Notable issues:
  - Setting the format of a table column on an imported TableWorkspace does nothing (I don't know if this used to work?)

**Comment by martyngigg** [comment]
These are internal changes so I'm happy that there are no user documentation updates.

**Comment by mantid-roman** [comment]
Column format used to work. Do you want me to fix it?

**Comment by martyngigg** [comment]
It currently works with a table created via "File->New Table" just not an imported table workspace, e.g. if I run a fit and double-click the parameters table. If that used to work then we should fix it in this pull request. If not then we can open another issue to fix it.

**Comment by mantid-roman** [comment]
I checked it used to work. I'll try to fix it in this PR.

**Comment by martyngigg** [comment]
The formatting seems on the imported `TableWorkspace` seems to work as expected now.

**Pull request merged**